.github/workflows/e2e.yaml

```diff
@@ -2,44 +2,42 @@ name: End-to-End Tests
 on:
   workflow_dispatch:
   pull_request:
-concurrency: build-wsl
+concurrency: end-to-end-tests
+
 jobs:
-  start_vm:
+  start-vm:
     strategy:
       max-parallel: 1
     runs-on: ubuntu-latest
     steps:
-      - uses: azure/login@v1
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Start Azure VM
+        uses: ./.github/actions/setup_vm/
         with:
-          creds: ${{ secrets.AZURE_VM_CREDS }}
-      - name: Start the Runner
-        run: |
-          az vm start --name wsl-ci --resource-group wsl
-  rdp:
+          az_name: wsl-ci
+          az_resource_group: wsl
+          az_creds: ${{ secrets.AZURE_VM_CREDS }}
+  
+  remote-desktop:
     strategy:
       max-parallel: 1
-    needs: start_vm
+    needs: start-vm
     runs-on: ubuntu-latest
     steps:
-      - run: |
-          set -o pipefail
-          sudo apt install -y xvfb freerdp2-x11
-          # detected empirically and confirmed here: https://github.com/FreeRDP/FreeRDP/blob/68ad8d5a1c8d52828095c2293acea1c76fa7fcb5/client/X11/xfreerdp.h#L350
-          vmOfflineErrorCode=141 
-          lastExitCode=$vmOfflineErrorCode
-          maxAttemptsCount=10;
-          while [[ $lastExitCode == $vmOfflineErrorCode && $maxAttemptsCount != 0 ]]; do
-            sleep 20 # seconds
-            xvfb-run -a xfreerdp /v:${{ secrets.AZURE_VM_AUTHORITY }} /u:${{ secrets.AZURE_VM_UN }} /p:'${{ secrets.AZURE_VM_UP }}' \
-              /cert:tofu | grep -v ${{ secrets.AZURE_VM_AUTHORITY }} || lastExitCode=$?
-            ((--maxAttemptsCount))
-          done
-            
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Connect via RDP
+        uses: ./.github/actions/rdp/
+        with:
+          az_vm_hostname: ${{ secrets.AZURE_VM_AUTHORITY }}
+          az_vm_username: ${{ secrets.AZURE_VM_UN }}
+          az_vm_password: ${{ secrets.AZURE_VM_UP }}
 
-  e2e:
+  end-to-end-tests:
     strategy:
       max-parallel: 1
-    needs: start_vm
+    needs: start-vm
     runs-on: self-hosted

```

Comment:
You lost me a little bit here:
1. this is not in the description of the PR
2. even with this, how does this job knows on which machine to run this? This is supposively the VM that we started but I don’t find any resource group or anything that clarify this… So, unsure to understand really (we can hop on a HO, maybe that would be clearer, but I think the documentation should help too)